helm-go-lib/main.go

```diff
@@ -174,6 +176,63 @@ func uninstallRelease(namespace string, kubeToken string, apiServer string, caFi
 	return C.CString("ok")
 }
 
+//export upgradeChart
+func upgradeChart(namespace string, kubeToken string, apiServer string, caFile string, releaseName string, chartName string, overrideValues string) *C.char {
+	actionConfig, err := buildActionConfig(namespace, kubeToken, apiServer, caFile)
+	if err != nil {
+		log.Printf("%+v\n", err)
+		return C.CString(err.Error())
+	}
+
+	client := action.NewUpgrade(actionConfig)
+	client.Namespace = namespace
+	client.CleanupOnFail = true
```

Comment:
Maybe add a comment about what `CleanupOnFail` does? Looking at https://helm.sh/docs/helm/helm_upgrade/:
```
--cleanup-on-fail                            allow deletion of new resources created in this upgrade when upgrade fails
```

Should this always be `true` or be a parameter to the function? 

Either way let's make triple sure that failed upgrades don't cause deletion of WDS [postgres deployment](https://github.com/broadinstitute/terra-helmfile/blob/master/charts/wds/templates/postgres.yaml#L49) (doesn't seem like it would because seems like it just deletes _new_ resources, but still..)

Reply:
If it indeed does what it says and delete only the NEW resources upon a failed upgrade, then I think setting it to True is a good option here.
But we better test it (have an existing app with existing resources, force fail the upgrade and make sure the original app is still there)